discussions/secure_web_api.md

```diff
@@ -0,0 +1,137 @@
+protected by vpn or PCI (Payment Card Information), and hence it may not be essential to enable mTLS in this case.
+
+However, in case of public network, clients will be communicating with L3AFD over insecure networks. L3AF wants to
+secure its endpoints using industry's best standard available solutions. This can be configured from l3afd’s config file,
+and by default, mTLS will be disabled.
+
+## Enabling mTLS for L3AFD
+
+L3AFD (L3AF Daemon) will require a set of root certificates (root.crt and root.key) and a pair of server certificates
+(server.crt and server.key) to run in mTLS enabled mode. L3AFD will check for these certificates, and in case these are
+not found, L3AFD will stop with error ```certificates are not found```.
+
+The client will require a pair of client certificates (client.crt and client.key), generated from the same root
+certificates from which the server certificates have been generated, and the public root certificate (root.crt) to
+communicate with L3AFD.
+
+Process to enable mTLS 
+- Provision of Certificates 
+- Location of Certificates 
+- Enabling mTLS
+
+### Provision of Certificates
+
+As mentioned before, users can use already pre-existing certificates. However, L3AF can also provide mechanisms
+manually to generate certificates.
+
+L3AF can provide a certificate generation tool can be implemented to generate the certificates manually.
+If the user has a set of root certificates (can be from a trusted CA provider or provided by its organisation or
+self-signed) and does not have the server and client certificates, then it can use the tool provided by L3AF to generate
+the server and client certificates. The tool can also have an option to generate the root certificates and the user can
+use this option to generate self-signed root certificates manually.
+
+#### Certificate creation tool options
+
+To create self-signed certificates, the tool needs following options and these can be passed as parameters.
+
+- Domain (by default, localhost or FQDN)
+- Password provided by user 
+- Message Digest (sha256 or sha512 or …)
+- Encryption algorithms (Camellia, Aria and AES is the popular one)
+- Subject (Country, State, Locality, Organisation, Organisation Unit and Common Name)
+  (e.g., “/C=IN/ST=KA/L=BGL/O=WM/OU=TBT/CN=$cn”)
+- Expiration days (365, 730 or 3652 days)
+- Alternate DNS, and IP (by default localhost, and 127.0.0.1)
+
+### Location of Certificates
+
+The default location for the certificates can be ‘/etc/ssl/l3af/certs/’ and this path can also be changed through a
+l3afd configuration option in cases where the user would like to use a custom location.
+
+The user will have to place the root certificates and the server certificates in the configured directory path before
+starting L3AFD.
+
+### Enabling mTLS
+
+L3AFD will provide a flag to enable mTLS and this can be configured in l3afd.cfg, by default mTLS will be disabled.
```

Comment:
I recommend switching the default so that you have to opt-in to be insecure.

Reply:
Initially, we considered enabling mTLS by default, however we decided against it since we believe most of them will use l3afd within their private networks.